src/components/Application.tsx

```diff
@@ -512,51 +523,63 @@ export class Application extends React.Component<Props, State> {
       isSearching: true,
       selectedFeature: null,
     })
+
     catalogService.search({
       count,
       startIndex,
       bbox: this.state.bbox,
       catalogApiKey: this.state.catalogApiKey,
       ...this.state.searchCriteria,
-    })
-      .then(searchResults => this.setState({ searchResults, searchError: null, isSearching: false }))
-      .catch(searchError => this.setState({ searchError, isSearching: false }))
+    }).then(searchResults => this.setState({
+      searchResults,
+      searchError: null,
+      isSearching: false,
+    })).catch(searchError => this.setState({
+      searchError,
+      isSearching: false,
+    }))
   }
 
   private handleSelectFeature(feature) {
     if (this.state.selectedFeature === feature) {
       return  // Nothing to do
     }
-    this.setState({
-      selectedFeature: feature || null,
-    })
+
     this.navigateTo({
       pathname: this.state.route.pathname,
-      search:   (feature && feature.properties.type === TYPE_JOB) ? `?jobId=${feature.id}` : '',
+      search: feature && feature.properties.type === TYPE_JOB ? `?jobId=${feature.id}` : '',
+      selectedFeature: feature,
     })
   }
 
   private navigateTo(loc) {
     const route = generateRoute(loc)
     history.pushState(null, null, route.href)
 
-    // Update selected feature if needed
-    let {selectedFeature} = this.state
+    // Update selected feature if needed.
+    let selectedFeature = 'selectedFeature' in loc
+      ? loc.selectedFeature
+      : this.state.selectedFeature
+
     if (route.jobIds.length) {
       selectedFeature = this.state.jobs.records.find(j => route.jobIds.includes(j.id))
     }
+
+    /* TODO: Okay?
```

Comment:
Is this TODO to check that it's ok to not reset to defaults everytime?

If so, that's ok, I think we want to just remember what they did last time